pkg/decoder/decoder.go

```diff
@@ -52,25 +74,55 @@ func (d *Decoder) Start() {
 	go d.run()
 }
 
+// Stop stops the Decoder
+func (d *Decoder) Stop() {
+	close(d.InputChan)
+}
+
 // run lets the Decoder handle data coming from the InputChan
 func (d *Decoder) run() {
 	for data := range d.InputChan {
-		d.decodeIncomingData(data.content, data.offset)
+		d.stopCountDown()
```

Comment:
I think we can come up with a better name than `stopCountDown`. Something like `ackIncomingData` maybe?

Reply:
Since we are just stoping the timer, I am not convinced